### PR TITLE
Enhances email formatting to avoid being flagged as spam

### PR DESCRIPTION
**Pull request by pcrespov**
## What do these changes do?

Following last spam issues reported by @oetiker (thx!). email plugin :
  - ensures that emails are rendered with a complete html
  - removes comments in templates used for testing (legacy)
  - cleans up ``templates/osparc/registration_email.jinja2`` html with https://www.htmlcorrector.com/ (and some manual touches) to comply with html standard
  - tests extra template rendering and html validation

## How to test

```cmd
cd services/web/server
make install-dev
pytest -vv tests/unit/**/test_*email*.py
```

### CODE REVIEW

**Review by oetiker — APPROVED** [review]
I like it :)

File: services/web/server/src/simcore_service_webserver/email_core.py [hunk 1]
```diff
@@ -183,6 +184,11 @@ async def send_email_with_attachements(
     return message
 
 
+def _remove_comments(html_string: str):
+    # WARNING: this function is mocked in the tests. Be careful when signature is changed.
```
Comment (GitHK) [inline services/web/server/src/simcore_service_webserver/email_core.py:188]: I would write a test to enforce this, since it seems important.

Comment (GitHK) [reply services/web/server/src/simcore_service_webserver/email_core.py:188]: Actually why not a test that uses this and enforces the comment is removed at this point?

**Review by sanderegg — APPROVED** [review]

**Review by mguidon — APPROVED** [review]
:+1:

**Review by GitHK — APPROVED** [review]
👍

**Comment by oetiker** [comment]
A further improvement would be to provide both a html and a text version of the mail (multipart alternative) this will further boost the legitimacy of the mail https://www.lifewire.com/email-marketing-multipart-alternative-1165136

**Comment by pcrespov** [comment]
> er improve

@oetiker thx for the input. Actually, multipart is also in place so this should be easy to add next iteration. Thx!

**Pull request merged**